### PR TITLE
Include cause in unexpected exceptions

### PR DESCRIPTION
**Pull request by peternied**
### Description
Include cause in unexpected exceptions

Also added `info` level logging to see these messages in a PR
And removed artifacts step as they were never used and the martix of jdk
release would override them (saves couple minutes from check runtime)

### Testing
Added an exception and verified it appeared in the output

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by cliu123** [comment]
The cause(error message) would be exposed to client. Is it sufficient to log the cause as a part of stacktrace in logs?

File: src/main/java/org/opensearch/security/filter/SecurityFilter.java [hunk 1]
```diff
@@ -373,7 +373,7 @@ public void onFailure(Exception e) {
             listener.onFailure(e);
         } catch (Throwable e) {
             log.error("Unexpected exception "+e, e);
-            listener.onFailure(new OpenSearchSecurityException("Unexpected exception " + action, RestStatus.INTERNAL_SERVER_ERROR));
+            listener.onFailure(new OpenSearchSecurityException("Unexpected exception " + action, RestStatus.INTERNAL_SERVER_ERROR, e));
```
Comment (cliu123) [inline src/main/java/org/opensearch/security/filter/SecurityFilter.java:376]: The cause(error message) would be exposed to client. Is it sufficient to log the cause as a part of stacktrace in logs?

Comment (peternied) [reply src/main/java/org/opensearch/security/filter/SecurityFilter.java:376]: That is a good call out, I'll create an issue test for that so we don't do this by accident.

Do you have a recommendation on how to make sure in the tests that we are getting the log output on line 375, it seems like in practice this isn't a real issue, but for our test cases we aren't seeing the expected results?

**Review by davidlago — APPROVED** [review]

**Comment by peternied** [comment]
Being that this PR was created to expose the exception message, and we shouldn't expose these messages in this way I have created https://github.com/opensearch-project/security/issues/1644 to track the root cause (tests aren't collecting/outputing log.error from the cluster under test).  Closing out this PR.

**Pull request closed without merging**